src/main/java/org/openhab/binding/zwave/ZWaveBindingConstants.java

```diff
@@ -89,14 +92,58 @@
     public final static String CHANNEL_CFG_BINDING = "binding";
     public final static String CHANNEL_CFG_COMMANDCLASS = "commandClass";
 
-    public final static String OFFLINE_CONTROLLER_OFFLINE_KEY = "zwave.thingstate.controller_offline";
-    public final static String OFFLINE_CONTROLLER_OFFLINE_DEFAULT = "Controller is offline";
-    public final static String OFFLINE_NODE_DEAD_KEY = "zwave.thingstate.node_dead";
-    public final static String OFFLINE_NODE_DEAD_DEFAULT = "Node is not communicating with controller";
-    public final static String OFFLINE_NODE_NOTFOUND_KEY = "zwave.thingstate.node_notfound";
-    public final static String OFFLINE_NODE_NOTFOUND_DEFAULT = "Node not found in Z-Wave network";
+    public final static I18nConstant OFFLINE_CTLR_OFFLINE = new I18nConstant("zwave.thingstate.controller_offline",
+            "Controller is offline");
+    public final static I18nConstant OFFLINE_NODE_DEAD = new I18nConstant("zwave.thingstate.node_dead",
+            "Node is not communicating with controller");
+    public final static I18nConstant OFFLINE_NODE_NOTFOUND = new I18nConstant("zwave.thingstate.node_notfound",
+            "Node not found in Z-Wave network");
+    public final static I18nConstant OFFLINE_SERIAL_EXISTS = new I18nConstant("zwave.thingstate.serial_notfound",
+            "Serial Error: Port {0} does not exist");
+    public final static I18nConstant OFFLINE_SERIAL_INUSE = new I18nConstant("zwave.thingstate.serial_inuse",
+            "Serial Error: Port {0} is in use");
+    public final static I18nConstant OFFLINE_SERIAL_UNSUPPORTED = new I18nConstant(
+            "zwave.thingstate.serial_unsupported", "Serial Error: Unsupported operation on port {0}");
+    public final static I18nConstant OFFLINE_SERIAL_LISTENERS = new I18nConstant("zwave.thingstate.serial_listeners",
+            "Serial Error: Too many listeners on port {0}");
 
     public final static Integer ACTION_CHECK_VALUE = new Integer(-232323);
 
     public final static Set<ThingTypeUID> SUPPORTED_BRIDGE_TYPES_UIDS = ImmutableSet.of(CONTROLLER_SERIAL);
+
+    private static I18nProvider i18nProvider;
+
+    protected void setI18nProvider(I18nProvider i18nProvider) {
+        ZWaveBindingConstants.i18nProvider = i18nProvider;
+    }
+
+    protected void unsetI18nProvider(I18nProvider i18nProvider) {
+        ZWaveBindingConstants.i18nProvider = null;
+    }
+
+    public static String getI18nConstant(I18nConstant constant) {
+        if (i18nProvider == null) {
+            return MessageFormat.format(constant.key, (Object[]) null);
+        }
+        return i18nProvider.getText(ZWaveActivator.getContext().getBundle(), constant.key, constant.defaultText, null,
```

Comment:
If you want to prevent a NPE in every case (I think you will perhaps never run into one), you have to change the code a little bit, because the unset method could be called async between the check and this line.